Self-service plan change documentation (#9603)
mendix/docs: content/en/docs/deployment/mendix-cloud-deploy/change-plan.md

{{% /alert %}}

## Introduction

Mendix offers several license plans including Free, Basic, Standard, Premium, and Premium Plus. For more information, see [Pricing](https://www.mendix.com/pricing/).

This document explains how to change your environment's [plan](/developerportal/deploy/mendix-cloud-deploy/#plans-1) in Mendix Cloud. Plan changes can be applied independently to each environment. For example, the Technical Contact can upgrade the application's acceptance environment while keeping production on its current plan.

{{% alert color="info" %}}
It is only possible to change and upgrade to higher-tier plans, not downgrade. Upgrades from legacy plans to new plans are supported, but not from new plans to legacy plans. 
{{% /alert %}}

## Prerequisites {#prerequisites}

Before requesting a plan change, verify the following:

* The app environment is deployed.
* The app runs on Kubernetes.
* The app's tenant is self-service enabled. To activate this capability for your company, contact your Customer Success Manager.
* You are the [Technical Contact](/support/#technical-contacts) for the app (only Technical Contacts can submit a plan change request).

> ## Changing Plan {#changing-plan}

Reviewer: I might say
```suggestion
## Changing a Plan {#changing-plan}
```

{{% alert color="info" %}}
This feature is only visible to the Technical Contact.
{{% /alert %}}

To change an environment's plan, follow these steps:

> 1. Go to [Apps](https://sprintr.home.mendix.com/) and click **Environments** on your app. 

Reviewer: I think this is three steps?

1. Go to Apps
2. Select your app
3. Click Environments.

Reviewer 2: Thanks for highlighting this. However, since Environments is visible from the app, I turned into 2 steps rather than 3.

2. On the **Environments** page, click the **Overview** tab to view your environments.
3. Find your target environment.
4. Use the **Expand** ({{< icon name="chevron-down" >}}) toggle to view detailed information about each environment, such as the environment resources and deployed package.
5. In the environment resource section, click **Change Plan**. 
6. In the **Plan Change Request** dialog that appears:

>     * On the left side, review your current plan details. This includes:

Reviewer: ```suggestion
    * On the left side, review your current plan details. These include:
```
?

>         * Plan **Name**.

Reviewer: In the documentation we tend to list the UX label and then describe it - like
```suggestion
        * **Name** – the plan name.
```
Does that make sense?

Reviewer 2: Yes! Fixed. Thanks

        * Plan **Environment** (such as, staging or acceptance).
        * **App** Name.
        * Plan **Memory**.
    * On the right side, create a new plan:
        * To compare plans, click **Show Comparison**. This opens a comparison page where the Technical Contact can evaluate the current plan against other available options. The last selected comparison will populate the form. Alternatively, select a new plan directly from the **Plan** dropdown without comparing.

>         * Enter a reason for plan change.

Reviewer: This seems to be actions, where the left side is just a description of the UX. I wonder if it would be better to have this as a separate step?

Reviewer 2: Yes! Thank you!!

> 7. Click **Submit Request** to submit your plan change request. After submission, the request appears in the [Control Center](/control-center/entitlements/#plan-change-requests-tab) for the [Mendix Admin](/control-center/company-settings/#mendix-admins) review. To view the list of approvers, click **Mendix Admin** at the bottom of the **New Plan** form. As a Technical Contact, you can [track and manage](#manage-plan-requests) your plan change requests from the [Request Overview](/developerportal/deploy/environments/#request-overview) tab on your app's environment's page. 

Reviewer: This seems to be multiple steps - submitting but then tracking. Perhaps they should be separate steps or at least make it clearer that we discuss track and trace later, perhaps in a separate paragraph?

Reviewer 3: Thanks for noting this. Changes made

Reviewer 2: Thanks for noting this. Changes made

> The app environment will remain on its current plan until the request is approved by a Mendix Admin.

Reviewer: I think we could be clearer that the Admin approval only allocates the new plan to the environment. The actual environment will remain on the current plan until the upgrade is scheduled and completed.

{{% alert color="info" %}}

> Changing plan consumes [Mendix Cloud Tokens](/control-center/entitlements/#cloud-tokens). Only one plan change request can exist per environment at a time. As a result, new requests cannot be submitted until the current one is completed.

Reviewer: It should be clear, but just for extra clarity I would say
```suggestion
Changing plan consumes [Mendix Cloud Tokens](/control-center/entitlements/#cloud-tokens). Only one plan change request can exist per environment at a time. As a result, new requests cannot be submitted for the same environment until the current one is completed.
```

* **Scheduled** – Click **Cancel Request** to cancel the entire request, or click **Execute Now** to apply the new plan immediately.

{{% alert color="warning" %}}
If the Technical Contact cancels a request, the plan change process is canceled entirely. To proceed with a plan change, the Technical Contact must start over and submit a new request.
{{% /alert %}}

## Scheduling a Plan Change {#scheduling-a-plan-change}

If your request status is **Pending Schedule**:

1. Open **Request Details** by clicking **Details** in the **Request Overview** tab.
2. Click **Schedule Change**.
3. Choose from the following options:
   * **Execute Now** to apply the change immediately
   * **Schedule for later** to delay the plan application until the next maintenance window

To schedule for later:

* Select a preferred date.
* The system will apply the change during the first available maintenance window after that date. For example, if you select May 22 and the next window is May 28, the change will occur on May 28.

> After scheduling, the status changes to **Scheduled**. You can still click **Execute Now** at any time before the scheduled window to apply the change immediately.

Reviewer: Does **Execute Now** override the maintenance window? I don't think it does, I think it just puts it in the next maintenance window. But perhaps it does act immediately - in which case I think we should mention that it doesn't use the maintenance window?

Reviewer 3: Yes, it overrides the maintenance window. Changes made to reflect that explicitly.

Reviewer 2: Yes, it overrides the maintenance window. Changes made to reflect that explicitly.

weight: 80
beta: true
description: "Describes how to change your plan in Mendix Cloud."

#Do NOT remove any of the anchors in this document as they are all referenced from other documents
---

{{% alert color="warning" %}}
This feature is in Public Beta. For more information, see [Beta Releases](/releasenotes/beta-features/).
{{% /alert %}}

## Introduction

Mendix offers several license plans including Free, Basic, Standard, Premium, and Premium Plus. For more information, see [Pricing](https://www.mendix.com/pricing/).

This document explains how to change your environment's [plan](/developerportal/deploy/mendix-cloud-deploy/#plans-1) in Mendix Cloud. Plan changes can be applied independently to each environment. For example, the Technical Contact can upgrade the application's acceptance environment while keeping production on its current plan.

{{% alert color="info" %}}
It is only possible to change and upgrade to higher-tier plans, not downgrade. Upgrades from legacy plans to new plans are supported, but not from new plans to legacy plans. 
{{% /alert %}}

## Prerequisites {#prerequisites}

Before requesting a plan change, verify the following:

> * The app environment is deployed.

Reviewer 4: This is not required. Environment can be in not deployed state.

1. Go to [Apps](https://sprintr.home.mendix.com/)
2. Click **Environments** on your app. 
3. On the **Environments** page, click the **Overview** tab to view your environments.
4. Find your target environment.
5. Use the **Expand** ({{< icon name="chevron-down" >}}) toggle to view detailed information about each environment, such as the environment resources and deployed package.
6. In the environment resource section, click **Change Plan**. 
7. On the left side of the form that appears, review your current plan details. These include:
    * **Name** – Plan name.
    * **Environment** – Plan environment (such as, staging or acceptance).
    * **App** – App name.
    * **Memory** – Allocated plan's memory.
8. On the right side of the form that appears, create a new plan:
    * To compare plans, click **Show Comparison**. This opens a comparison page where the Technical Contact can evaluate the current plan against other available options. The last selected comparison will populate the form. Alternatively, select a new plan directly from the **Plan** dropdown without comparing.
    * Enter a reason for plan change.
9. Click **Submit Request** to submit your plan change request. 

> 10. Submitted requests appear in the [Control Center](/control-center/requests/) for the [Mendix Admin](/control-center/company-settings/#mendix-admins) review. To view the list of approvers, click **Mendix Admin** at the bottom of the **New Plan** form.

Reviewer 4: I think we can mention this before asking to submit, that request will go to company admins for review and you can check the admin list by clicking Mendix Admins.

Reviewer 2: Thanks for the review. Now implemented.
